src/trie.py

```diff
@@ -46,3 +46,25 @@ def contains(self, token):
             else:
                 return False
         return True
+
+    def traversal(self, start):
```

Comment:
Per the assignment specs, and in the interests of general usability, your `traversal` method should not require any positional arguments. The user doesn't necessarily know what starting places are available in the top-level dictionary.

This will require some refactoring, but it will likely make your `traversal` method simpler.